examples/serve/auth-server.yaml

```diff
@@ -0,0 +1,10 @@
+service:
+  readiness_probe: /
+  auth_key: sky-authkey-3d2105b9-a9ba-4f13
```

Comment:
Can it take an `env_var` instead? Also, should we instead allow the `readiness_probe` to specify a header as well, so the user can specify the authentication in the header directly, instead of adding a new field `auth_key`?

Reply:
The ability to add headers is implemented in #3552, but that is for the authorization for replica endpoint. Do you mean we should infer the authorization for LB from replica directly?
(We could discuss more in our next meeting

Reply:
Ahh, this PR is for adding authorization for the load balancer along. We should check if this is necessary with users. : )

Reply:
SG! Lets defer it